code/addons/docs/src/preset.ts

```diff
@@ -194,4 +194,12 @@ const docsX = docs as any;
 
 ensureReactPeerDeps();
 
-export { webpackX as webpack, indexersX as experimental_indexers, docsX as docs };
+const optimizeDeps = [
+  '@mdx-js/react',
+  '@storybook/addon-docs > acorn-jsx',
+  '@storybook/addon-docs',
+  '@storybook/addon-essentials/docs/mdx-react-shim',
+  'markdown-to-jsx',
+];
```

Comment:
Moved from the Vite builder's default optimization, as that would break if `addon-essentials` was installed without `react`, which is a valid scenario now.